lock-api/src/test/java/com/palantir/lock/client/LeasingTimelockClientTest.java

```diff
@@ -0,0 +1,222 @@
+    }
+
+    @Test
+    public void leasedTokenShouldHaveValidLeaseForTheLeasePeriod() {
+        LockRequest lockRequest = mock(LockRequest.class);
+        when(timelockService.leasableLock(lockRequest)).thenReturn(
+                LeasableLockResponse.of(LOCK_RESPONSE, getLease()));
+
+        LockResponse lockResponse = timelockClient.lock(lockRequest);
+        assertValid(lockResponse.getToken());
+    }
+
+    @Test
+    public void unlockShouldCallRemoteServer_validLeases() {
+        LeasedLockToken token = LeasedLockToken.of(LOCK_TOKEN, getLease());
+        assertValid(token);
+        timelockClient.unlock(ImmutableSet.of(token));
+
+        verify(timelockService).unlock(ImmutableSet.of(token.serverToken()));
+    }
+
+    @Test
+    public void unlockShouldCallRemoteServer_inValidLeases() {
+        LeasedLockToken token = LeasedLockToken.of(LOCK_TOKEN, getLease(Duration.ZERO));
+        assertInvalid(token);
+        timelockClient.unlock(ImmutableSet.of(token));
+
+        verify(timelockService).unlock(ImmutableSet.of(token.serverToken()));
+    }
+
+    @Test
+    public void unlockShouldInvalidateLease() {
+        LockToken token = LeasedLockToken.of(LOCK_TOKEN, getLease());
+        timelockClient.unlock(ImmutableSet.of(token));
+
+        assertInvalid(token);
+    }
+
+    @Test
+    public void shouldOnlyCallIdentifiedTimeIfLeaseIsValid() {
+        LeasedLockToken validToken = LeasedLockToken.of(LOCK_TOKEN, getLease());
+        when(timelockService.getLeaderTime()).thenReturn(getIdentifiedTime());
+        timelockClient.refreshLockLeases(ImmutableSet.of(validToken));
+
+        verify(timelockService).getLeaderTime();
+        verifyNoMoreInteractions(timelockService);
+    }
+
+    @Test
+    public void shouldRefreshTheLease_invalidOnClient_validOnServer() {
+        LeasedLockToken leasedLockToken = LeasedLockToken.of(LOCK_TOKEN, getLease(Duration.ZERO));
+        assertInvalid(leasedLockToken);
+
+        when(timelockService.leasableRefreshLockLeases(ImmutableSet.of(LOCK_TOKEN)))
+                .thenReturn(LeasableRefreshLockResponse.of(
+                        ImmutableSet.of(LOCK_TOKEN),
+                        getLease()
+                ));
+
+        Set<LockToken> refreshed = timelockClient.refreshLockLeases(ImmutableSet.of(leasedLockToken));
+        verify(timelockService).leasableRefreshLockLeases(ImmutableSet.of(leasedLockToken.serverToken()));
+
+        LeasedLockToken refreshedLeasedLockToken = (LeasedLockToken) refreshed.iterator().next();
+        assertValid(refreshedLeasedLockToken);
+        assertValid(leasedLockToken);
+        assertEquals(leasedLockToken, refreshedLeasedLockToken);
+    }
+
+    private StartIdentifiedAtlasDbTransactionResponse startTransactionResponseWith(LockToken lockToken) {
+        return StartIdentifiedAtlasDbTransactionResponse.of(
+                LockImmutableTimestampResponse.of(1L, lockToken),
+                mock(TimestampAndPartition.class)
+        );
```

Comment:
prefer not putting braces on the next line; style within AtlasDB (and at Palantir in general) to close braces all at once.

Reply:
same in the other cases